ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetTransactionCountTest.java

```diff
@@ -56,4 +75,107 @@ public void verifyTransactionCount() {
 
     assertThat(response.getResult()).isEqualTo(String.format("0x%X", NONCE));
   }
+
+  @Test
+  public void validRequestProducesExpectedNonce() {
+    final PrivateTransactionHandler privateNonceProvider = mock(PrivateTransactionHandler.class);
+    when(privateNonceProvider.getSenderNonce(any(), any())).thenReturn(NONCE);
+
+    final PrivacyGroup[] privacyGroups =
+        new PrivacyGroup[] {
+          new PrivacyGroup(
+              "Group1", PrivacyGroup.Type.PANTHEON, "Group1_Name", "Group1_Desc", new String[0]),
+          new PrivacyGroup(
+              "Group2", PrivacyGroup.Type.LEGACY, "Group2_Name", "Group2_Desc", new String[0]),
+        };
+    final Enclave enclave = mock(Enclave.class);
+    when(enclave.findPrivacyGroup(any())).thenReturn(privacyGroups);
+
+    final PrivGetTransactionCount method =
+        new PrivGetTransactionCount(parameters, privateNonceProvider, enclave);
+
+    final JsonRpcResponse response = method.response(legacyRequest);
+    assertThat(response).isInstanceOf(JsonRpcSuccessResponse.class);
+
+    final JsonRpcSuccessResponse successResponse = (JsonRpcSuccessResponse) response;
+    final int returnedValue = Integer.decode((String) successResponse.getResult());
+    assertThat(returnedValue).isEqualTo(NONCE);
+  }
+
+  @Test
+  public void nonceProviderThrowsRuntimeExceptionProducesErrorResponse() {
+    final PrivateTransactionHandler privateNonceProvider = mock(PrivateTransactionHandler.class);
+    when(privateNonceProvider.getSenderNonce(any(), any())).thenThrow(RuntimeException.class);
+
+    final PrivacyGroup[] privacyGroups =
```

Comment:
Looks like the test data for privacy groups is the same for all tests. Can this be just initialised once, perhaps in the setup?

Reply:
Created constants for the three different privacy groups used in the tests